Userland/Libraries/LibWeb/Page/EditEventHandler.cpp

```diff
@@ -111,8 +111,15 @@ void EditEventHandler::handle_insert(JS::NonnullGCPtr<DOM::Position> position, u
         builder.append(node.data().bytes_as_string_view().substring_view(0, position->offset()));
         builder.append_code_point(code_point);
         builder.append(node.data().bytes_as_string_view().substring_view(position->offset()));
-        node.set_data(MUST(builder.to_string()));
 
+        // Cut string by max length
+        auto new_string = MUST(builder.to_string());
+        auto new_string_view = new_string.bytes_as_string_view();
+        if (node.max_length().has_value() && new_string_view.length() > *node.max_length()) {
+            node.set_data(MUST(String::from_utf8(new_string_view.substring_view(0, min(new_string_view.length(), *node.max_length())))));
+        } else {
+            node.set_data(new_string);
+        }
```

Comment:
This isn't quite right because the length is supposed to be measured by the UTF-16 code unit length of the string, not the byte length. So let's at least add a FIXME for that.

We can also do this less allocate-y, and that `min` doesn't seem needed since we know the string length is larger than the max length.

Let's also use `String::substring` instead of going from String -> StringView -> StringView -> String
```c++
if (auto max_length = node.max_length(); max_length.has_value() && builder.string_view().length() > *max_length) {
    node.set_data(MUST(String::from_utf8(builder.string_view().substring_view(0, *max_length))));
} else {
    node.set_data(MUST(builder.to_string()));
}
```